### PR TITLE
Modify `CONTRIBUTING.md` and the issue templates `bug_report.yml` and `feature_request.yml` to reduce duplicate issues and improve the quality of further submitted issues

### LINKED ISSUE #56007: Search with search engine to reduce duplicate issues
### Acknowledgement

- [X] I acknowledge that issues using this template may be closed without further explanation at the maintainer's discretion.

### Comment

**Problem:** You and I create duplicate issues not because one didn‘t search, but because the GitHub search engine sucks. Its search algorithm is defective that the actually relevant search results most often appear on page 5 or something because synonyms are not generally considered.

**Solution:** Mention in both the CONTRIBUTING.md and the issue templates that, instead of searching in GitHub, search for `site:github.com/microsoft/TypeScript <your keywords>` in your favorite search engine. In most cases, relevant results should appear on the top even if one is not sure and doesn‘t include the correct terminology.

### LINKED ISSUE #58206: Modify the issue templates `bug_report.yml` and `feature_request.yml` to improve the quality of further submitted issues
### Acknowledgement

- [X] I acknowledge that issues using this template may be closed without further explanation at the maintainer's discretion.

### Comment

Per [Ryan’s comment](https://github.com/microsoft/TypeScript/pull/58155#issuecomment-2057520029), here are my proposed changes to the issue templates `bug_report.yml` and `feature_request.yml`:

- Move the guidances up in both templates to make the text slightly bigger and clearer
- Add a reminder to read CONTRIBUTING.md before opening an issue in both templates
- Add a line about the number of opening bugs in `bug_report.yml` to parallelize with `feature_request.yml`
- Remove the duplicated “List/Write them (down) here so that others can find this bug more easily” in `bug_report.yml`

You may find a preview of the changes in #58155 (I am not going to copy the `diff` here) and the previews of the modified templates here:

- [`bug_report.yml`](https://github.com/graphemecluster/TypeScript/issues/new?assignees=&labels=&projects=&template=bug_report.yml)
- [`feature_request.yml`](https://github.com/graphemecluster/TypeScript/issues/new?assignees=&labels=&projects=&template=feature_request.yml)

I try to be helpful rather than making a nuisance of myself. If I am doing so by opening this issue, please kindly inform me and close it.

### PR DESCRIPTION
**Pull request by graphemecluster**
Fixes #56007
Fixes #58206

### CODE REVIEW

**Review by RyanCavanaugh — CHANGES_REQUESTED** [review]
This has a bunch of extraneous changes that weren't described in the linked bug

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 1]
```diff
@@ -3,17 +3,24 @@ description: 'Create a report to help us improve TypeScript'
 body:
   - type: markdown
     attributes:
-      value: Please fill in each section completely. Thank you!
+      value: |
+        💡 Did you know? TypeScript has over 1,500 opening bug reports!
```
Comment (RyanCavanaugh) [inline .github/ISSUE_TEMPLATE/bug_report.yml:7]: Remove

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 2]
```diff
@@ -3,17 +3,24 @@ description: 'Create a report to help us improve TypeScript'
...
+
+        📖 Please read [CONTRIBUTING.md](https://github.com/microsoft/TypeScript/blob/main/CONTRIBUTING.md) comprehensively before opening an issue.
```
Comment (RyanCavanaugh) [inline .github/ISSUE_TEMPLATE/bug_report.yml:9]: Remove

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 3]
```diff
@@ -3,17 +3,24 @@ description: 'Create a report to help us improve TypeScript'
...
     id: search_terms
     attributes:
       label: '🔎 Search Terms'
       description: |
         What search terms did you use when trying to find an existing bug report?
 
-        List them here so people in the future can find this one more easily.
-      placeholder: |
-        List of keywords you searched for before creating this issue. Write them down here so that others can find this bug more easily and help provide feedback.
+        List them here so people in the future can find this bug report more easily and help provide feedback.
```
Comment (RyanCavanaugh) [inline .github/ISSUE_TEMPLATE/bug_report.yml:23]: Undo

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 4]
```diff
@@ -3,19 +3,26 @@ description: 'Suggest an idea'
 body:
   - type: markdown
     attributes:
-      value: 'Please fill in each section completely. Thank you!'
+      value: |
+        💡 Did you know? TypeScript has over 2,000 open suggestions!
```
Comment (RyanCavanaugh) [inline .github/ISSUE_TEMPLATE/feature_request.yml:7]: Same here, undo unrelated changes

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -10,6 +10,8 @@ Issues that ask questions answered in the FAQ will be closed without elaboration
 
 [Search the existing issues](https://github.com/Microsoft/TypeScript/search?type=Issues) before logging a new one.
 
+You should also search for `site:github.com/microsoft/TypeScript <your keywords>` in your favorite **search engine**. In most cases, you should find the results listed at the top more relevant and accurate than the searching feature in GitHub.
```
Comment (RyanCavanaugh) [inline CONTRIBUTING.md:13]: Combine with prior sentence; this doesn't need its own paragraph

**Comment by graphemecluster** [comment]
I am willing to make such modifications, but are there any particular reasons why these changes would not be desirable? Are they having adverse effects?

**Comment by RyanCavanaugh** [comment]
> I am willing to make such modifications, but are there any particular reasons why these changes would not be desirable? Are they having adverse effects?

Including unrelated changes in a PR is an adverse effect on its own terms. If you want to change other aspects of the template, file another bug.

**Comment by graphemecluster** [comment]
I guess the alterations should be something acceptable now

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 6]
```diff
@@ -3,7 +3,12 @@ description: 'Create a report to help us improve TypeScript'
 body:
   - type: markdown
     attributes:
-      value: Please fill in each section completely. Thank you!
+      value: |
+        🔍 Please [search thoroughly in GitHub](https://github.com/Microsoft/TypeScript/search?type=Issues) or by the query `site:github.com/microsoft/TypeScript <your keywords>` in your favorite **search engine** before reporting a new bug as most bugs are very likely to find precedents.
+
+        Besides searching in GitHub, you should also search for `site:github.com/microsoft/TypeScript <your keywords>` in your favorite **search engine** for more relevant and accurate results.
```
Comment (RyanCavanaugh) [inline .github/ISSUE_TEMPLATE/bug_report.yml:9]: This sentence is the same content as the sentence before it.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -8,7 +8,7 @@ Issues that ask questions answered in the FAQ will be closed without elaboration
 
 ## 2. Search for Duplicates
 
-[Search the existing issues](https://github.com/Microsoft/TypeScript/search?type=Issues) before logging a new one.
+[Search the existing issues in GitHub](https://github.com/Microsoft/TypeScript/search?type=Issues) or by the query `site:github.com/microsoft/TypeScript <your keywords>` in your favorite **search engine** before logging a new one. Search engines generally list more relevant and accurate results at the top than the GitHub searching feature.
```
Comment (RyanCavanaugh) [inline CONTRIBUTING.md:11]:
```suggestion
[Search the existing issues in GitHub](https://github.com/Microsoft/TypeScript/search?type=Issues) or by the query `site:github.com/microsoft/TypeScript <your keywords>` in your favorite search engine before logging a new one. Search engines generally list more relevant and accurate results at the top than the GitHub searching feature.
```

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 6 above)
Comment (RyanCavanaugh) [reply .github/ISSUE_TEMPLATE/bug_report.yml:9]: Also, un-emph "search engine"

**Review by RyanCavanaugh — CHANGES_REQUESTED** [review]

**Review by RyanCavanaugh — APPROVED** [review]
Thanks!
